walletdatabase/migrations/sql/1704701942_remove_favourite_and_change_primary_key_for_saved_addresses.up.sql

```diff
@@ -0,0 +1,42 @@
+CREATE TABLE IF NOT EXISTS saved_addresses_new (
+  address VARCHAR NOT NULL,
+  name VARCHAR NOT NULL DEFAULT "",
+  removed BOOLEAN NOT NULL DEFAULT FALSE,
+  update_clock INT NOT NULL DEFAULT 0,
+  chain_short_names VARCHAR NOT NULL DEFAULT "",
+  ens_name VARCHAR NOT NULL DEFAULT "",
+  is_test BOOLEAN NOT NULL DEFAULT FALSE,
+  created_at INT NOT NULL DEFAULT 0,
+  color VARCHAR NOT NULL DEFAULT "primary",
+  PRIMARY KEY (address, is_test),
+  UNIQUE (name, is_test)
```

Comment:
I see test mode polluting more of the functionality by adding extra responsibilities, I'm thinking that we should consider abstracting it at a higher level until it is not too entangled. My first thought is that having a separate wallet DB for testing might make it easier.

Reply:
Not sure if you're pointing to status-go unit/integration tests or test mode, in this example `is_test` field is used to mark the addresses that are saved when the app is in the test mode.

Reply:
> Not sure if you're pointing to status-go unit/integration tests or test mode, in this example is_test field is used to mark the addresses that are saved when the app is in the test mode.
I was referring to the blockchain test mode (where we use the chains instead of .

> Not entirely sure up to which level is the test mode entangled in the app at this moment, like do we differentiate the data for all wallet-related stuff based on the test mode, but if we do, that's a valid suggestion.

We have to use it for all user state-related data that should not be mixed between main/test modes (usually chain-related like in this case). We should keep an eye on it though if it grows too much.